File: PYTHON/Lista/Lista.py
class Lista:
    def __init__(self):
        self.paises = []

    def agregar(self,pais):
        self.paises.append(pais)

    def buscar(self,pais):
        for i in self.paises:
            if pais:
                return True
        return False

    def ordenar(self):
        self.paises.sort()

    def desordenar(self):
        self.paises.sort(reverse=True)

    def eliminar(self,pais):
        self.paises.remove(pais)

    def insertar(self,posicion,pais):
        self.paises.insert(posicion,pais)

    def recorridoInicioFin(self):
        print(self.paises)

    def recorridoFinInicio(self):
        print(self.paises[::-1])

    def imprimir(self):
        print(self.paises)

File: C#/Lista/Lista.cs
using System;
using System.Collections.Generic;

namespace Lista
{
    public class Lista
    {
        private List<string> lista;
        public Lista()
        {
            this.lista = new List<string>();
        }

        public void agregar(string pais)
        {
            lista.Add(pais);
        }

        public string buscar(string pais)
        {
            return lista.Find(x => x == pais);
        }

        public void ordenar()
        {
            lista.Sort();
        }

        public void desordenar()
        {
            lista.Reverse();
        }

        public void eliminar(string pais)
        {
            lista.Remove(pais);
        }

        public void insertar(int pos, string pais)
        {
            lista.Insert(pos, pais);
        }

        public void recorridoInicioFin()
        {
            foreach (string item in lista)
            {
                Console.WriteLine(item);
            }
        }

        public void recorridoFinInicio()
        {
            List<string> aux = lista;
            aux.Reverse();
            foreach (string item in aux)
            {
                Console.WriteLine(item);
            }
        }

        public void imprimir()
        {
            foreach (string item in lista)
            {
                Console.WriteLine(item);
            }
        }
    }
}